xtokens/src/mock/mod.rs

```diff
@@ -338,11 +339,60 @@ impl WeightTrader for AllTokensAreCreatedEqualToWeight {
 		Ok(unused)
 	}
 
-	fn refund_weight(&mut self, weight: Weight) -> Option<MultiAsset> {
+	fn refund_weight(&mut self, weight: Weight, _context: &XcmContext) -> Option<MultiAsset> {
 		if weight.is_zero() {
 			None
 		} else {
 			Some((self.0.clone(), weight.ref_time() as u128).into())
 		}
 	}
 }
+
+/// Allows execution from all origins taking payment into account.
+///
+/// Only allows for `TeleportAsset`, `WithdrawAsset`, `ClaimAsset` and
+/// `ReserveAssetDeposit` XCMs because they are the only ones that place assets
+/// in the Holding Register to pay for execution. This is almost equal to
+/// [`xcm_builder::AllowTopLevelPaidExecutionFrom<T>`] except that it allows for
+/// multiple assets and is not generic to allow all origins.
+pub struct AllowTopLevelPaidExecution;
```

Comment:
Add comment explain why this is duplicated and link to upstream issue https://github.com/paritytech/polkadot-sdk/issues/1638

Reply:
done 👍